haiku/src/implementation.cc

```diff
@@ -0,0 +1,60 @@
+#include "iana-time-zone-haiku/src/interface.h"
+#include "iana-time-zone-haiku/src/lib.rs.h"
+
+#ifdef __HAIKU__
+
+#include <cstring>
+
+#include <Errors.h>
+#include <LocaleRoster.h>
+#include <String.h>
+#include <TimeZone.h>
+
+size_t ::iana_time_zone_haiku::get_tz(rust::Slice<uint8_t> buf) {
+    try {
+        static_assert(sizeof(char) == sizeof(uint8_t), "Illegal char size");
+
+        if (buf.empty()) {
+            return 0;
+        }
+
+        BLocaleRoster *locale_roster(BLocaleRoster::Default());
```

Comment:
Can you add a comment here saying that `BLocaleRoster::Default()` returns a static pointer and thus no free is necessary?

Reply:
The [documentation](https://www.haiku-os.org/docs/api/classBLocaleRoster.html#a7cb5f911107993a9c4816b52e9221572) is not terribly explicit about what to do with the returned pointer, but at the place of the implementation [a comment](https://github.com/haiku/haiku/blob/8f16317a5b6db5c672f331814273e5857555020f/src/kits/locale/LocaleRoster.cpp#L143-L147) indicates that the pointer is static. → [`MutableLocaleRoster::Default()`](https://github.com/haiku/haiku/blob/8f16317a5b6db5c672f331814273e5857555020f/src/kits/locale/MutableLocaleRoster.cpp#L59-L66)